test/WebSites/XmlFormattersWebSite/Controllers/ValidationController.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNet.Mvc;
+using Microsoft.AspNet.Mvc.ModelBinding;
+
+namespace XmlFormattersWebSite
+{
+    public class ValidationController : Controller
+    {
+        public IActionResult CreateStore([FromBody] Store store)
+        {
+            // We want to verify that 'store' is model bound and also that the
+            // model state has the errors we are expecting.
+            return new ObjectResult(new ModelBindingInfo()
+            {
+                Store = store,
+                ModelStateErrorMessages = GetModelStateErrorMessages(ModelState)
+            });
+        }
+
+        private List<string> GetModelStateErrorMessages(ModelStateDictionary modelStateDictionary)
```

Comment:
good question!...also this found an issue in my PR where i am supposed to add the model state errors as exceptions in order to not reveal the details to end user (i had this earlier but somehow changed it)...

So to answer your question, since SerializableError does not reveal details of the model state errors having exceptions, I needed to do this explicitly to be able to verify the error messages at the client. I will add a comment here for clarity.